float64 rover_px
float64 rover_py
float64 rover_vx
float64 rover_vy
float64 rover_w
int32 rover_state